compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangFiniteTypeNode.java

```diff
@@ -79,14 +79,21 @@ public NodeKind getKind() {
 
     @Override
     public String toString() {
-        StringJoiner stringJoiner = new StringJoiner(" | ");
+        StringJoiner stringJoiner = new StringJoiner("|");
         for (BLangExpression memberTypeNode : valueSpace) {
-            if (memberTypeNode.getBType().tag == TypeTags.FLOAT) {
-                stringJoiner.add(memberTypeNode.toString() + NumericLiteralSupport.FLOAT_DISCRIMINATOR);
-            } else if (memberTypeNode.getBType().tag == TypeTags.DECIMAL) {
-                stringJoiner.add(memberTypeNode.toString() + NumericLiteralSupport.DECIMAL_DISCRIMINATOR);
-            } else {
-                stringJoiner.add(memberTypeNode.toString());
+            switch (memberTypeNode.getBType().tag) {
+                case TypeTags.FLOAT:
+                    stringJoiner.add(memberTypeNode + NumericLiteralSupport.FLOAT_DISCRIMINATOR);
+                    break;
+                case TypeTags.DECIMAL:
```

Comment:
Shall we write a test to cover these lines?

Reply:
I will look into this.